### PR TITLE
fix: bug where appointments can appear outside the calendar

### PR DESCRIPTION
**Pull request by NathanBP**
- Fixes bug #1203. Appointments set for 11:59pm now stay inside the calendar boundary
- Added the dndresource demo to the demo drop down so that it can be viewed. 
- Updated resources in the dndresource demo to show bug has been fixed.

### CODE REVIEW

File: src/utils/TimeSlots.js [hunk 1]
```diff
@@ -130,7 +130,10 @@ export function getSlotMetrics({ min: start, max: end, step, timeslots }) {
 
       const rangeStartMin = positionFromDate(rangeStart)
       const rangeEndMin = positionFromDate(rangeEnd)
-      const top = (rangeStartMin / (step * numSlots)) * 100
+      const top =
+        rangeStartMin === rangeEndMin
+          ? (rangeStartMin / (step * numSlots)) * 98
```
Comment (jquense) [inline src/utils/TimeSlots.js]: what's going on here with `98`, is there a deterministic reason to use 98%? This feels like a number that would only work with the default slot numbers.

Comment (NathanBP) [reply src/utils/TimeSlots.js]: Yes you are correct after further testing with smaller step values it does make the events not sit in the correct space. I have modified the code for this section to factor in the step value so as to allow this to now work correct for any step value.

File: test/utils/TimeSlots.test.js [hunk 2]
```diff
@@ -22,3 +22,81 @@ describe('getSlotMetrics', () => {
...
+    expect(range.top + range.height).toBeLessThan(100)
+    expect(range.height).toBeGreaterThan(0)
+  })
+
+  test('getRange: multi day appointment stays within calendar', () => {
+    let range = slotMetrics.getRange(
+      new Date(2018, 0, 29, 0, 0, 0),
+      new Date(2018, 0, 30, 4, 0, 0)
+    )
+    expect(range.top + range.height).toBeLessThan(100)
```
Comment (jquense) [inline test/utils/TimeSlots.test.js]: Should this be equal to 100?

Comment (NathanBP) [reply test/utils/TimeSlots.test.js]: I believe so, as there is code in the getRange that selects the max value between the start of the day and the start of the appointment. Along with the start of the day and the end of the appointment. My understanding is that this is what splits a multi day appointment over separate days in the week view when the `showMultiDayTimes` prop is set. 

I might be wrong though as this is just my understanding of how the calendar displays appointments

**Comment by NathanBP** [comment]
> Looks great, is it worth adding a test covering a start date before the time slot range as well?

I thought this scenario would have been covered by having a test that is from 00:00 to 00:15 and the multi day test? I can add another test if you'd like, if possible could you clarify further by giving an example date/time?

**Comment by NathanBP** [comment]
Just a quick comment to keep this pull request open

**Comment by jquense** [comment]
thanks!

**Pull request merged**